### PR TITLE
Add sparse data support for LogisticRegression

### PR DESCRIPTION
**Pull request by avolkov-intel**
### Description

This PR introduce changes to support sparse data for Logistic Regression training and inference on GPU (for CPU sparse data will not be supported yet). Also a few tests were implemented

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Review by Vika-F — COMMENTED** [review]
The changes look good to me. Even versioning is in place.
For some reason I cannot see the results of pre-commit testing, but if everything is Ok there, then LGTM.

File: onedal/linear_model/logistic_regression.cpp [hunk 1]
```diff
@@ -41,7 +41,10 @@ struct method2t {
 
         const auto method = params["method"].cast<std::string>();
         ONEDAL_PARAM_DISPATCH_VALUE(method, "dense_batch", ops, Float, method::dense_batch);
-        ONEDAL_PARAM_DISPATCH_VALUE(method, "by_default", ops, Float, method::dense_batch);
+        #if defined(ONEDAL_VERSION) && ONEDAL_VERSION >= 20240700
+        ONEDAL_PARAM_DISPATCH_VALUE(method, "sparse", ops, Float, method::sparse);
+        #endif // defined(ONEDAL_VERSION) && ONEDAL_VERSION >=20240700
```
Comment (Alexsandruss) [inline onedal/linear_model/logistic_regression.cpp:46]: Wrong formatting
```suggestion
#if ONEDAL_VERSION >= 20240700
        ONEDAL_PARAM_DISPATCH_VALUE(method, "sparse", ops, Float, method::sparse);
#endif
```

File: onedal/linear_model/tests/test_logistic_regression.py [hunk 2]
```diff
@@ -64,3 +65,30 @@ def test_pickle(queue, dtype):
         result = model2.predict(X, queue=queue)
 
         assert_array_equal(expected, result)
+
+
+if daal_check_version((2024, "P", 600)):
```
Comment (Alexsandruss) [inline onedal/linear_model/tests/test_logistic_regression.py:70]: This condition mentions 2024.6.0 version while `onedal` and onther places - 2024.7.0. Are you sure you set correct versions?

**Review by Alexsandruss — CHANGES_REQUESTED** [review]

**Review by Alexsandruss — APPROVED** [review]

**Comment by avolkov-intel** [comment]
@Mergifyio copy rls/2024.7.0-rls

**Pull request merged**